include/aws/auth/credentials.h

```diff
@@ -266,6 +266,32 @@ struct aws_credentials_provider_sts_options {
     aws_io_clock_fn *system_clock_fn;
 };
 
+/**
+ * The process credentials provider sources credentials from running a command or process.
+ * The command is configured in config file with the config key as "credential_process".
```

Comment:
I'd reword this along the lines of:

"The command to run is sourced from a profile in the AWS config file, using the standard profile selection rules.  The profile key the command is read from is "credential_process." 

And then maybe give a very simple example ala

[default]
credential_process=/usr/local/bin/my-credential-fetcher --stage=Prod